conn.go

```diff
@@ -35,21 +38,70 @@ type Conn interface {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+type wrappedNetConn struct {
+	net.Conn
+	bytesRead int
+	addr      net.Addr
+}
+
+var _ net.Conn = new(wrappedNetConn)
+
+func (c *wrappedNetConn) Read(b []byte) (int, error) {
```

Comment:
The idea behind this wrapper is that the bufio.Reader will make subsequent calls to Read (within `resp3.Unmarshal`) until it has all the data it needs for a particular message to be read. Since this wrapped Read can extend its read deadline automatically it greatly reduces the chances of getting partial reads, which we can't recover from.